src/adapters/search_history.js

```diff
@@ -0,0 +1,41 @@
+import { get, set, del } from './store';
+
+const SEARCH_HISTORY_KEY = 'search_history';
+const HISTORY_SIZE = 100;
+
+// Add a query to the list
+export function saveQuery(q) {
+  // Delete query if it's already in the list
+  deleteQuery(q);
+
+  // Retrieve the search history
+  const searchHistory = get(SEARCH_HISTORY_KEY) || [];
+
+  // Put the query at the end of the array
+  searchHistory.push(q);
+
+  // Limit the list to the 100 last items
+  if (searchHistory.length > HISTORY_SIZE) {
+    searchHistory.shift();
```

Comment:
`shift` is a bit fragile, as it won't react properly if we happen to reduce HISTORY_SIZE.
We can get the n last elements of the array with a negative argument passed to `slice`, which will work in all cases.